packages/riverpod/test/providers/provider/provider_test.dart

```diff
@@ -35,6 +35,20 @@ void main() {
       expect(container.read(provider), 1);
     });
 
+    test(
+      'can be refreshed even if the type of state is nullalbe AsyncValue',
```

Comment:
> I'm not really sure this is something that I want to officially support.
> 
> Support for this will likely drop with metaprogramming

Oh I see... Thanks for your comment!